backend/network.py

```
@@ -0,0 +1,175 @@
+        if self._tunnel_host:
+            self._tunnel()
+        self.sock = self._context.wrap_socket(
+            self.sock, server_hostname=self._tls_server_hostname
+        )
+
+
+def _port(parts: urllib.parse.SplitResult) -> int:
+    try:
+        return parts.port or (443 if parts.scheme == "https" else 80)
+    except ValueError:
+        raise errors.blocked_url("invalid port")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)